### ISSUE TITLE
weights of arcface module not updating

### ISSUE THREAD

**Issue by wfcb-85** [TreB1eN/InsightFace_Pytorch#71]
Hi,
After each epoch, I print the kernel variable of the ArcFace Module, but its weights are always the same although the gradients are not zero. 
Should not the model head be set to train() when training ?

**Comment by graycrown** [comment]
I have met the same problem, did you solve it?
My training loss only decress in first step, after that, it alway keep the same value like this:
tensor(34.6456, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
tensor(3.6976, device='cuda:0', grad_fn=<NllLossBackward>)
If  you has any suggestion, plz tell me , I'll be grateful
